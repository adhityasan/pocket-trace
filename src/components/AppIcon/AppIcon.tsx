import MaskedView from "@react-native-masked-view/masked-view";
import { useMemo } from "react";
import { Image, ImageStyle, StyleProp, View } from "react-native";

import { COLOR_LIGHT } from "@services/constants/color";

interface IAppIconProps {
  size?: number;
  color?: string;
  style?: StyleProp<ImageStyle>;
}

const AppIcon = ({ size = 50, color = COLOR_LIGHT.PRIMARY, style }: IAppIconProps) => {
  const imageStyle = useMemo(() => {
    const removedBackgroundColor = { ...(style as { [keys: string]: any }) };
    delete removedBackgroundColor.backgroundColor;
    return removedBackgroundColor;
  }, [style]);

  return (
    <MaskedView
      maskElement={
        <Image
          style={[{ width: size, height: size }, imageStyle]}
          source={require("../../assets/images/app-icon.png")}
        />
      }>
      <View style={[{ backgroundColor: color, width: size, height: size }, style]} />
    </MaskedView>
  );
};

export default AppIcon;
