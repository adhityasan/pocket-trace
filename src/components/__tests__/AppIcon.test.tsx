import { create } from "react-test-renderer";

import AppIcon from "@components/AppIcon";

describe("components/AppIcon", () => {
  it("renders correctly", () => {
    const tree = create(<AppIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("has mask image & masked view", () => {
    expect.assertions(3);
    const tree: any = create(<AppIcon />).toJSON();
    const hasMaskImage = tree.children[0].children[0].type === "Image";
    const hasMaskedView = tree.children[1].type === "View";
    expect(tree.children.length).toBe(2);
    expect(hasMaskImage).toBeTruthy();
    expect(hasMaskedView).toBeTruthy();
  });
});
