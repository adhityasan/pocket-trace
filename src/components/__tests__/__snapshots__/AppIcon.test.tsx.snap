// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AppIcon renders correctly 1`] = `
<RNCMaskedView>
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 50,
            "width": 50,
          },
          Object {},
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#5145a0",
          "height": 50,
          "width": 50,
        },
        undefined,
      ]
    }
  />
</RNCMaskedView>
`;
